auth/src/main/java/org/aerogear/auth/utils/AuthStateManager.java

```diff
@@ -0,0 +1,60 @@
+package org.aerogear.auth.utils;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import org.aerogear.auth.credentials.OIDCCredentials;
+import org.json.JSONException;
+
+/**
+ * Saves, retrieves and delete a token.
+ */
+public class AuthStateManager {
+
+    private static final String STORE_NAME = "org.aerogear.android.auth.AuthState";
+    private static final String KEY_STATE = "state";
+
+    private SharedPreferences mPrefs;
+
+    public AuthStateManager(Context context) {
+        this.mPrefs = context.getSharedPreferences(STORE_NAME, Context.MODE_PRIVATE);
+    }
+
+    /**
+     * Reads credentials from storage.
+     * @return OIDCCredentials
+     */
+    public OIDCCredentials read() {
+        String currentState = mPrefs.getString(KEY_STATE, null);
+        if (currentState == null) {
+            return new OIDCCredentials();
+        }
+        try {
+            return new OIDCCredentials(currentState);
+        } catch (JSONException ex) {
+            return new OIDCCredentials();
+        }
+    }
+
+    /**
+     * Saves a token
+     * @param authState token to be saved
+     */
+    public void write(OIDCCredentials authState) {
+        if (authState == null) {
+            clear();
+        } else {
+            if(!mPrefs.edit().putString(KEY_STATE, authState.serialise()).commit()) {
+                throw new IllegalStateException("Failed to update state from shared preferences");
+            }
+        }
+    }
+
+    /**
+     * Deletes a token
+     */
+    public void clear() {
```

Comment:
also to support multiple threads to read and write the auth state, you might want to consider using a lock for write and clear operations.

Reply:
@wei-lee I've declared the methods synchronized in this commit https://github.com/aerogear/aerogear-android-sdk/pull/13/commits/736d254e4834dbd93a02054d95e04971174c9465